numba/targets/cpu.py

```diff
@@ -74,6 +75,9 @@ def load_additional_registries(self):
         self.install_registry(randomimpl.registry)
         self.install_registry(randomimpl.registry)
 
+        # load 3rd party extensions
+        numba.entrypoints.init_all()
```

Comment:
I think this being called here (the main caller of `load_additional_registries` being `context.refresh()` ?) breaks the semantics of a refresh as the entry point initialisation system has a global state https://github.com/numba/numba/pull/4427/files#diff-bc21c945c4d5ed7fe59fdd6af1a44d37R5.

Reply:
Is there a better place to put the load?  I'm not sure I understand the issue.  I don't want the extensions to be loaded repeatedly, hence the flag check.

Reply:
For now, I'm not sure there is a better place. My concern is that because there is state, if an extension needs to talk to the Numba state, a refresh of one doesn't trigger the other. Perhaps we leave it as is and see what happens in practice with view of revisiting as needs be.